src/routes/rooms/update-room-by-id.ts

```diff
@@ -0,0 +1,67 @@
+import { Prisma } from '@prisma/client';
+import { Response } from 'express';
+import { prisma } from '../../prisma';
+import { RequestWithUser } from '../../types/request-with-user';
+import { paramIdValidationSchema } from './param-id.validation';
+import { roomNameValidatorSchema } from './room.validation';
+
+export const updateRoomNameById = async (
+  req: RequestWithUser,
+  res: Response
+) => {
+  const id = paramIdValidationSchema.safeParse(req.params?.id);
+  const body = roomNameValidatorSchema.safeParse(req.body);
+  const userId = req.user?.id;
+
+  if (!id.success) {
+    return res.status(400).json({ success: false, message: 'Invalid param' });
+  }
+
+  if (!body.success) {
+    return res
+      .status(400)
+      .json({ success: false, message: 'Invalid room request body' });
+  }
+
+  const room = await prisma.room.findUnique({
+    where: {
+      id: id.data,
+    },
+    select: {
+      id: true,
+      creatorId: true,
+    },
+  });
+
+  if (room === null) {
+    return res
+      .status(400)
+      .json({ success: false, message: 'Room does not exist' });
+  }
+
+  if (room.creatorId !== userId) {
+    return res
+      .status(403)
+      .json({ success: false, message: 'Player is not the owner of the room' });
+  }
+
+  try {
+    await prisma.room.update({
+      where: {
+        id: id.data,
+      },
+      data: {
+        name: body.data.name,
+      },
+    });
+  } catch (err) {
```

Comment:
Aqui o seu código continua até retornar `{ success: true }` se o `error.code` não for `P2002`. Recomendo retornar um `500` caso o erro não caia em nenhum `if`.

Reply:
certo, irei adicionar um tratamento!